airflow/cli/cli_config.py

```diff
@@ -793,6 +814,13 @@ def string_lower_type(val):
     action="store_true",
 )
 
+# IMPORTANT NOTE!
+#
+# Celery configs below have explicit fallback values because celery provider defaults are not yet loaded
+# via provider at the time we parse the command line, so in case it is not set, we need to have manual
+# fallback
+# DO NOT REMOVE THE FALLBACKS even if you are tempted to.
+# TODO: possibly move the commands to providers but that could be big performance hit on the CLI
```

Comment:
@potiuk 

AIP-51 PR https://github.com/apache/airflow/pull/29055 will move the commands to the executor modules (which will be in providers now) so this will actually be done by that.

I've spent time optimizing the import times of the executor modules so it should not slow down the CLI much at all.

Was there another concern you had with regards to performance?

Reply:
There is one concern I have, after my chnages, some of the commands (celery, kubernetes ) will have to initialize ProvidersManager's initialize_configuration() - which is going to be costly and it cannot be done during parsing of the arguments, so I am afraid extracting them to providers are not going to be "really" possible. I think we will have to eventually leave the CLI definition for Celery and Kubernetes commands in the core because of that. But I suggest we wait for it until we see it, we can likely have it separated out, but not moved to inside provider's code.

Reply:
I have added test/protection to see if there is an accidental usage of providers manager while running CLI parsing. I think it might be useful once you will be merging #29055 - see  https://github.com/apache/airflow/pull/32694

I think with the right lazy loading approach it should be possible to the the #29055 as well. But let's see.